### ISSUE TITLE
Fullscreen toggling on Windows broken (RetroArch issue)

### ISSUE THREAD

**Issue by loganmc10** [loganmc10/GLupeN64#80]
I've changed Windows to use the OpenGL core profile. Unfortunately this means that fullscreen toggling causes RetroArch to crash.

Something is wrong with the Core Profile on Windows (with RetroArch), this works without issue in Linux, so there is something going wrong inside RetroArch.

**Comment by SigmaVirus** [comment]
What does all this mean? I thought GLupeN64 was already using OpenGL? What's different from how it were before?

**Comment by loganmc10** [comment]
OpenGL can operate in 2 modes: Core and Compatibility.

Compatibility mode basically says "allow me to use old / deprecated code from OpenGL 2". Core mode basically says "I promise I'm only going to use proper modern OpenGL 3.2+ code". 

Some drivers (Mac OS X and open source Linux graphics drivers) only allow Core mode by default. This is the reason GLideN64 has this issue: https://github.com/gonetz/GLideN64/issues/454

GLupeN64 uses a modified version of GLideN64 that removes the deprecated OpenGL code and supports OpenGL core, so those issues don't exist on Linux for us.

Anyway basically GLupeN64 supports the Core profile so that is what it should use. RetroArch Windows doesn't handle full screen toggling for some reason. It even has some issues with Compatibility mode (try full screen toggling with an older GLupeN64 build using retroarch_debug.exe and it will also crash) I need to get on IRC and sort out with the RetroArch devs why this is happening

**Comment by loganmc10** [comment]
I've decided I don't have the energy to sort this out with RetroArch so I'll revert that change (b1726009e939c1dcccf444e3beeb92e72f9a71fd).

I still don't like it though, it should be using the Core profile, but I suppose functionally should probably come before correctness.

**Comment by SigmaVirus** [comment]
@loganmc10 
There's no shame in reverting commits, I understand now what you were trying to achieve.
You meant well, but I can assure you that there's plenty of time to deal with this later on.

And as you said, functionality before correctness and this is something I think all of us end users agree with.
Besides all I care for right now is a functional N64 emulator that does it's job and GLupeN64 is just that!

I'm an layman so OpenGL 2 or OpenGL 3.2+ doesn't really mean much to me as long as the games work.

**Comment by orbea** [comment]
I would not be surprised if this was a windows issue rather than RetroArch issue...
